src/EditorFeatures/Core.Wpf/InlineRename/InlineRenameSession.cs

```diff
@@ -741,11 +762,25 @@ private void ApplyRename(Solution newSolution, IWaitContext waitContext)
                         var newText = newDocument.GetTextAsync(waitContext.CancellationToken).WaitAndGetResult(waitContext.CancellationToken);
                         finalSolution = finalSolution.WithDocumentText(id, newText);
                     }
+
+                    // Make sure to include any document rename as well
+                    finalSolution = finalSolution.WithDocumentName(id, newDocument.Name);
                 }
 
                 if (_workspace.TryApplyChanges(finalSolution))
                 {
-                    if (!_renameInfo.TryOnAfterGlobalSymbolRenamed(_workspace, changedDocumentIDs, this.ReplacementText))
+                    // Since rename can apply file changes as well, and those file 
+                    // changes can generate new document ids, include added documents
+                    // as well as changed documents. This also ensures that any document
+                    // that was removed is not included
+                    var finalChanges = _workspace.CurrentSolution.GetChanges(_baseSolution);
```

Comment:
not certain i understand this.  but i don't really know the contract that is expected.  If you+jason are ok with this, then 'm ok.

Reply:
The document id gets changed with a rename, so we want to make sure we're reporting that correctly. I talked with @jasonmalinowski 	 about this, and believe this is the most correct we can be.